core/deployment/src/main/java/io/quarkus/artemis/core/deployment/DevServicesArtemisProcessor.java

```diff
@@ -72,6 +73,17 @@ public List<DevServicesResultBuildItem> startArtemisDevService(
             DevServicesConfig devServicesConfig) {
         ArrayList<DevServicesResultBuildItem> results = new ArrayList<>();
         for (String name : bootstrap.getConfigurationNames()) {
+            String propertyName = name;
+            for (String rawName : ConfigProvider.getConfig().getPropertyNames()) {
+                if (rawName.contains(name)) {
+                    String str = rawName.substring("quarkus.artemis.".length());
+                    str = str.substring(0, str.indexOf('.'));
```

Comment:
This is not going to work. In theory, a name like `quarkus.artemis."my.connection"...` is legal. If I understand the algorithm correctly, `str` would be `my`, while the `propertyName`  would be `my.connection`.

If we want to do it properly, I fear we have to do some parsing...

@radcortez WDYT?

Reply:
I spoke with @radcortez and he recommends using `io.smallrye.config.NameIterator` to analyze the config name.

Reply:
Yeah, I will try to use `NameIterator`.

Reply:
Also `"foo.'1'"` is valid property name?

Reply:
Yes, `NameIterator` removes quotes from the segments unless they are escaped.

Reply:
@zhfeng the `NameIterator` should provide access to the offsets of a segment. So to figure out whether the name is quoted, we'd need to check the characters before and after the segment.

Reply:
Thanks all - I update the algorithm to use `nameIterator`, please take a review again!